geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/GCStatsMonitor.java

```diff
@@ -19,68 +19,82 @@
 import org.apache.geode.internal.statistics.StatisticNotFoundException;
 import org.apache.geode.internal.statistics.StatisticsNotification;
 
+/**
+ * This class acts as a monitor and listen for GC statistics updates on behalf of MemberMBean.
+ * <p>
+ * There's only one dedicated sampler thread that mutates the fields and writes the statistics to a
+ * file. The mutable fields are declared as {@code volatile} to make sure readers of the statistics
+ * get the latest value recorded.
+ * <p>
+ * The class is not thread-safe. If multiple threads access an instance concurrently, it must be
```

Comment:
You should make clear that it is safe to have multiple concurrent readers of this class.
It is only the methods that modify (handleNotification, decreasePrevValues, and increaseStats) that need external synchronization.